### PR TITLE
make: rename AD to Q

### PR DESCRIPTION
**Pull request by kaspar030**
The variable "AD" was used to add "@" to makefile recipe lines depending on the QUIET setting.

Does anyone remember why the variable was called like this?

Anyhow, this PR changes the variable name to a short "Q".

### CODE REVIEW

**Comment by miri64** [comment]
`AD` is called `AD` for the Latin ad which ist what "@" stands for. ;)

**Comment by miri64** [comment]
Why Q?

**Comment by kaspar030** [comment]
quiet :)

**Comment by gebart** [comment]
Q as in quiet.
FWIW, the makefiles Contiki also use a variable named Q for the quiet @

**Comment by miri64** [comment]
But you put the `Q` at the position where the `@` usually would be

**Comment by kaspar030** [comment]
I don't read AD as @ and make the connection...

Anyhow, I want to introduce a "quiet when building for CI" variable, and IMO "CI_AD" is even more confusing than "CI_Q".

**Comment by haukepetersen** [comment]
+1 for the `Q`.

> AD is called AD for the Latin ad which ist what "@" stands for. ;)

nice to know - I always thought somebody made a funny mistake :-)

**Comment by x3ro** [comment]
Can't we just write `QUIET` and avoid future confusion? 🎉 Kind of hard to search for a single letter in any codebase.

**Comment by gebart** [comment]
No for anything long like QUIET. This is a prefix variable that will be written on every line of the makefile recipes, so it should be short and out of the way.

**Comment by kYc0o** [comment]
+1 for Q, I was also confused but then I remove it and see what happened xD

**Comment by haukepetersen** [comment]
I am with @gebart here.

**Comment by kaspar030** [comment]
> Can't we just write QUIET and avoid future confusion?

QUIET can be used by the user to set AD.

**Comment by smlng** [comment]
>
>Does anyone remember why the variable was called like this?
>
I can image a future PR saying the same for 'Q'; renaming it again to something different (or back to 'AD'. 

I don't see much benefit in this other than @kaspar030 gaining one or two commits 😄

**Comment by kaspar030** [comment]
> I don't see much benefit in this other than @kaspar030 gaining one or two commits

> Anyhow, I want to introduce a "quiet when building for CI" variable, and IMO "CI_AD" is even more confusing than "CI_Q".

The idea is to not have to grep out all the "make -C blah" lines *after* the CI system has transmitted them (20k times...), but keep them for normal use.

I'm leaning towards calling the second variable "QQ" (double quiet). With AD, I don't know how to name it.

**Comment by kaspar030** [comment]
- rebased
- added QQ variable

@smlng Does the rename make more sense for you like this?
@miri64 How about you?

File: Makefile.include [hunk 1]
```diff
@@ -83,10 +83,12 @@ endif
 QUIET ?= 1
 
 ifeq ($(QUIET),1)
-  AD=@
+  Q=@
+  QQ=@
```
Comment (smlng) [inline Makefile.include]: I thought this should only apply for CI builds?

**Comment by smlng** [comment]
@kaspar030 you got me wrong -- I have no bet on either naming. I was just questioning the reasoning behind and if it is really that important, or just leave as is.

However, I do understand the need for _quiet_ builds in general and for CI in particular, but why an extra `QQ` instead of extending `AD` (or `Q`, what ever the name will be) to those steps as well.

IMHO it makes less sense to have `quiet` and `more quiet`, instead it would be more appropriate to have:
 
- sensible default output, 
- a _quiet_ option that suppresses all (or most) output, and
- maybe a _verbose_ option (`V`?) showing everything?

```
QUIET ?= 1
VERBOSE ?= 0

ifeq ($(QUIET),1)
Q=@
V=@
else ifeq($(VERBOSE),1)
Q=
V=
else
Q=
V=@
endif
```

**Comment by kaspar030** [comment]
- addressed @smlng's comment

**Comment by kaspar030** [comment]
> it would be more appropriate to have

Maybe. But it's a lot more intrusive to change.

**Comment by kaspar030** [comment]
> IMHO it makes less sense to have quiet and more quiet, instead it would be more appropriate to have:
> 
> sensible default output,
> a quiet option that suppresses all (or most) output, and
> maybe a verbose option (V?) showing everything?

In essence, we have sensible default output, and verbose (QUIET=0), and "make &> /dev/null".

I just want a way to not have to clumsily "grep -v 'make -C'" from the build output.
@smlng Maybe we go with this PR for now and, if deemed necessary, improve later? The change is rather minimal...

**Comment by kaspar030** [comment]
@cgundogan @smlng What do you think? should we make result output more readable?

**Comment by kaspar030** [comment]
> @cgundogan @smlng What do you think? should we make result output more readable?

*ping*

Please take a look at https://ci.riot-labs.de/devel/RIOT-OS/RIOT/6390/162e1b8179ee9e0f8971e1026bb2035777989898/output.html. More than two thirds of RIOT's build output consists of those make lines...

**Comment by smlng** [comment]
@kaspar030 I cannot recall what you mean with more readable, I'm fine with changes in this PR, if that's the question. So I'll approve ...

**Comment by kaspar030** [comment]
@smlng thx! &go.

**Pull request merged**